test-resources/openApiSpecs/events-private-spec.yaml

```
@@ -0,0 +1,213 @@
+            X-Frame-Options:
+              schema:
+                type: "string"
+        # "401":
+        #   description: Unauthorized - sessionId does not exist
+        #   headers:
+        #     Access-Control-Allow-Origin:
+        #       schema:
+        #         type: "string"
+        #     Cache-Control:
+        #       schema:
+        #         type: "string"
+        #     Content-Type:
+        #       schema:
+        #         type: "string"
+        #     Strict-Transport-Security:
+        #       schema:
+        #         type: "string"
+        #     X-Content-Type-Options:
+        #       schema:
+        #         type: "string"
+        #     X-Frame-Options:
+        #       schema:
+        #         type: "string"
+        "500":
+          description: >-
+            Internal server error - NB: This endpoint is implemented as an APIGW service proxy - consumers should be aware
+            that the underlying service may generate 5XX errors due to the distributed nature of AWS therefore
+            occasional 5XX responses are safe to retry - see
+            [AWS documentation on retries](https://docs.aws.amazon.com/general/latest/gr/api-retries.html)
+          headers:
+            Access-Control-Allow-Origin:
+              schema:
+                type: "string"
+            Cache-Control:
+              schema:
+                type: "string"
+            Content-Type:
+              schema:
+                type: "string"
+            Strict-Transport-Security:
+              schema:
+                type: "string"
+            X-Content-Type-Options:
+              schema:
+                type: "string"
+            X-Frame-Options:
+              schema:
+                type: "string"
+      x-amazon-apigateway-integration:
+        type: "aws"
+        passthroughBehavior: "never"
+        httpMethod: "POST"
+        uri: "arn:aws:apigateway:eu-west-2:dynamodb:action/Query"
+        credentials:
+          Fn::GetAtt: ["EventsApiRole", "Arn"]
+        requestTemplates:
+          application/json:
+            Fn::Sub:
+              - >
+                #set($inputRoot = $input.params('$'))
+                #set($session = $input.params('session'))
+                #set($eventName = $input.params('eventName'))
+                {
+                  "TableName": "${tableName}",
+                  "ExpressionAttributeValues": {
+                    ":pk": {
+                      "S": "SESSION#$session"
+                    },
+                    ":skPrefix": {
+                      "S": "TXMA#EVENT_NAME#$eventName"
+                    }
+                  },
+                  "ExpressionAttributeNames": {
+                    "#pk": "pk",
+                    "#sk": "sk",
+                    "#eb": "eventBody"
+                  },
+                  "KeyConditionExpression": "#pk = :pk AND begins_with(#sk, :skPrefix)",
+                  "ProjectionExpression": "#eb"
```

```suggestion
                  "ProjectionExpression": "#event"
```
This depends on what we've called the Dynamo property. This will also need updating in the ExpressionAttributeNames

---

Updated